### PR TITLE
Use mock-etcd for cargo test.

### PR DESCRIPTION
**Pull request by pzheng1025**

### CODE REVIEW

File: csi/build.rs [hunk 1]
```diff
@@ -8,5 +8,5 @@ fn main() {
         "src",                                    // output
         None,                                     // customizations
     )
-    .expect("Failed to compile gRPC definitions!");
+    .unwrap_or_else(|_| panic!("Failed to compile gRPC definitions!"));
```
Comment (pwang7) [inline csi/build.rs:11]: do not ignore the error
`|err| panic!("Failed to compile gRPC definitions, the error is: {}", e)`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -135,13 +129,12 @@ jobs:
...
       #    sudo chmod u+s $BIND_MOUNTER
       #    ls -lsh $BIND_MOUNTER
-      - name: Restart etcd service to clean up its data
-        run: |
-          docker kill $ETCD_CONTAINER_NAME
-          docker run -d --rm --net host --name $ETCD_CONTAINER_NAME gcr.io/etcd-development/etcd:latest
-          docker ps
-          docker logs $ETCD_CONTAINER_NAME
-          sudo netstat -lntp
+      - name: Run etcd service
```
Comment (pwang7) [inline .github/workflows/ci.yml:132]: you don't need to run the etcd docker

Comment (pzheng1025) [reply .github/workflows/ci.yml:150]: This is needed by sanity test.

File: csi/build.rs (hunk 1 above)
Comment (pzheng1025) [reply csi/build.rs]: Done

**Review by pwang7 — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -71,6 +71,33 @@ jobs:
...
+          command: build
+          args: --bin bind_mounter
+      - name: Enable bind_mounter permission and run cargo test
+        run: |
+          sudo chown root:root $BIND_MOUNTER
+          sudo chmod u+s $BIND_MOUNTER
+          ls -lsh $BIND_MOUNTER
+          export ETCD_END_POINT=http://127.0.0.1:2379
+          export BIND_MOUNTER=`realpath $BIND_MOUNTER`
+          cargo test -p csi
```
Comment (pwang7) [inline .github/workflows/ci.yml:100]: also run cargo test with root to test direct mount, see
https://github.com/datenlord/datenlord/blob/f1240e3ffc3cd6516b034749d3e8216bbbc84465/.github/workflows/ci.yml#L72

**Review by pwang7 — CHANGES_REQUESTED** [review]

**Review by pwang7 — APPROVED** [review]

**Pull request merged**